### ISSUE TITLE
wsl.exe Wrong Default Display Langauge

### ISSUE THREAD

**Issue by mdragosv** [microsoft/WSL#9242]
### Version

Microsoft Windows [Version 10.0.22621.900]

### WSL Version

- [ ] WSL 2
- [ ] WSL 1

### Repro Steps

Install Windows 11 with the romanian language pack, maybe other languages affected.
Run wsl --install
Do wsl --help the help language wont be in english as expected

![image](https://user-images.githubusercontent.com/29517827/205050465-58bf2713-69ef-41ed-887b-73707ba59580.png)

### Expected Behavior

Default to english

### Actual Behavior

PS C:\Users\marin> wsl -v
Verze WSL: 1.0.0.0
Verze jádra: 5.15.74.2
Verze WSLg: 1.0.47
Verze MSRDC: 1.2.3575
Verze Direct3D: 1.606.4
Verze DXCore: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
verze Windows: 10.0.22621.900
PS C:\Users\marin>
![image](https://user-images.githubusercontent.com/29517827/205050766-c3ac18ac-350c-49bb-9e97-5d8611bd00c2.png)

### Solution ###

https://github.com/microsoft/WSL/issues/9242#issuecomment-1406138325

**Comment by benhillis** [comment]
Could you try `powershell Get-WinSystemLocale`?

**Comment by mdragosv** [comment]
> Could you try `powershell Get-WinSystemLocale`?

LCID             Name             DisplayName
----             ----             -----------
1048             ro-RO            Romanian (Romania)
![image](https://user-images.githubusercontent.com/29517827/205459291-de976c14-80b1-4c0f-99bb-86d02f86af1e.png)

**Comment by Nuklon** [comment]
I have a similar problem, although it goes to Dutch. My display language is English:
```
LCID             Name             DisplayName
----             ----             -----------
1033             en-US            English (United States)
```

I don't even have Dutch installed.

**Comment by Sedemicra** [comment]
Had a similar issue, where WSL display language was in Czech. 

In language settings:
Windows display language: English (United States)
Under Preferred languages: Only Estonian (language pack not even downloaded), which also set Apps & websites language to Estonian.

Solution:
Installed English (United States) language pack under Preferred languages. Left the order remain (Estonian first, English second), so the Apps & websites language remained Estonian, but WSL diaplay language turned to English (as desired).

Hope it is helpful for others dealing with similar issue.

**Comment by dorinlazar** [comment]
Same issue here, only the system locale is en-US. However, I do have the country and region set to Romania and the default input method set as Romanian (Programmers).
![image](https://user-images.githubusercontent.com/11272/219433017-fba08d37-20f4-4157-b522-bc865d627242.png)

**Comment by mdragosv** [comment]
> Had a similar issue, where WSL display language was in Czech.
> 
> In language settings: Windows display language: English (United States) Under Preferred languages: Only Estonian (language pack not even downloaded), which also set Apps & websites language to Estonian.
> 
> Solution: Installed English (United States) language pack under Preferred languages. Left the order remain (Estonian first, English second), so the Apps & websites language remained Estonian, but WSL diaplay language turned to English (as desired).
> 
> Hope it is helpful for others dealing with similar issue.

Can confirm your solution works works

**Comment by mdragosv** [comment]
> Same issue here, only the system locale is en-US. However, I do have the country and region set to Romania and the default input method set as Romanian (Programmers). ![image](https://user-images.githubusercontent.com/11272/219433017-fba08d37-20f4-4157-b522-bc865d627242.png)

Add English United States as additional language and it should display

![image](https://user-images.githubusercontent.com/29517827/219727232-e0483949-7053-4e2e-a802-403e85a24e1c.png)

**Comment by dorinlazar** [comment]
Still, it's a problem that should be clarified - is the Romanian language package filled with Czech localization strings, or is there a bug in wsl?

**Comment by Sedemicra** [comment]
@dorinlazar In my case there was nothing to do with Romanian language. I myself suspect WSL, but being not familiar in detail, how languages and locales work, can't really confirm my suspicions.

**Comment by mdragosv** [comment]
> 

Bug in WSL

**Comment by Stanzilla** [comment]
My Windows is set to English and there is no language pack installed:

- The CLI output for commands is in *English*
- The clie output for the `update` command is in *German*
which leads me to believe it uses some sort of GEO IP or the region settings? Either way, pretty weird, would love to have everything in English.

**Comment by gril002** [comment]
It's been almost a year and it's still broken. @benhillis Can we expect a solution anytime soon?

**Comment by JEAYNE** [comment]
Same problem here since the **upgrade** from WSL 1.2.5.0 to 2.0.7.0

My Windows 11 is setup to use English for the Language and French for the Region.
```
PS> systeminfo | findstr Locale
System Locale:             en-gb;English (United Kingdom)
Input Locale:              en-gb;English (United Kingdom)

PS> Get-WinSystemLocale

LCID             Name             DisplayName
----             ----             -----------
2057             en-GB            English (United Kingdom)

PS> xyz
xyz : The term 'xyz' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
```

It looks like that  sometime WSL.EXE uses the **Region**, and sometime it uses the **Language**.

For some options WLS.EXE displays result in French (my Windows Region):

```
PS> wsl.exe --version
Version WSL : 2.0.7.0
Version du noyau : 5.15.133.1-1
Version WSLg : 1.0.59
Version MSRDC : 1.2.4677
Version direct3D : 1.611.1-81528511
Version de DXCore : 10.0.25880.1000-230602-1350.main
Version de Windows : 10.0.22621.2428
```

For other options it displays result in English (my Windows Language):

```
PS> wsl -l -v
  NAME                      STATE           VERSION
* Alpine                    Running         2
  docker-desktop            Stopped         2
  docker-desktop-data       Stopped         2
  OracleLinux_7_9           Stopped         2
  podman-machine-default    Stopped         2
```

I guess everyone expect an output in the same language as the Language selected in the 'Language & Region' settings.

**Comment by AndreasBrostrom** [comment]
This just suddenbly happend for me today:
![image](https://github.com/microsoft/WSL/assets/13811113/fe1c415e-ee89-4a9e-b915-7565f35513f0)
![image](https://github.com/microsoft/WSL/assets/13811113/89d8570c-542b-4a47-a63a-a69549439be7)

**Comment by dstarkowski** [comment]
Single command - two languages.

![image](https://github.com/microsoft/WSL/assets/11463380/1f9fa230-22b5-420c-9aad-1e5969c3a9d0)

It's a widespread problem across Microsoft products where Culture and UI Culture are mixed.

Region/Regional format/Location should never affect the language. Either Windows display language or the first one on preferred language list should be used (these are both en-US for me).

**Comment by danielniccoli** [comment]
Why on earth is my WSL German? I installed Windows in *English (United States)*.

![grafik](https://github.com/microsoft/WSL/assets/2971735/a59641a4-6ad5-4408-ab7a-9cffcee4a476)

![grafik](https://github.com/microsoft/WSL/assets/2971735/b6dc292d-9d1d-46db-aa30-c4c34a425d7f)

**Comment by iburakov** [comment]
I have the same problem right after doing the `wsl --update` from 1.x.x.x to 2.0.14.0. WSL output language has changed to Russian.
![image](https://github.com/microsoft/WSL/assets/5229130/5e22928f-466f-4e8e-8ea8-e8c961f2b7ec)

Should've been English - the chosen Windows display language and the topmost preferred language:
![image](https://github.com/microsoft/WSL/assets/5229130/2f87ecf9-d2f3-4292-9b2f-f1f4d0116e93)

Mentioned solutions don't work. Chosen regional date/time format should NOT affect the WSL output language.

**Comment by romasandu-gaijin** [comment]
After some experimentation, it is clear that the language WSL uses for localization is taken from the "reginal format" setting, as several people have pointed out above, which is 100% a bug. People can and do use other contries formats without knowing the language, and not being able to even read the help without changing system settings to something inconvenient is awful. Please fix.

**Comment by stefan-oskamp** [comment]
Can confirm that wsl.exe 2.0.14 is switching its display language when you change the Windows Region and Regional Format.

What part of "Windows Display Language = English (US)" does wsl.exe not understand??? Does it need more AI or maybe just more common sense?

**Comment by stianthaulow** [comment]
Changing Regional format does indeed change the wsl language:
![wsl-region](https://github.com/microsoft/WSL/assets/43840258/0c37507a-f39c-4e80-961b-3fd5913be7fb)

WSL version: 2.0.14.0
Kernel version: 5.15.133.1-1
WSLg version: 1.0.59
MSRDC version: 1.2.4677
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.19045.3930

**Comment by danielniccoli** [comment]
Can this please be labeled as a bug?

**Comment by danielniccoli** [comment]
> Please, know, that advanced users and especially developers and administrators usually never want a translated version.

I disagree.  If applications are translated, they should come in the language the user expects. You're using a generalisation to express your personal opinion. That does not help the quality of the application and only shifts the issue to a different group of people - namely the ones who want localised apps. If **you** want your system and apps to be in English, then you can do that via the system language settings.

The problem here is that the application does **not** follow the system language, but rather the region. These are two different settings. A user can be located in Germany, Poland, France or any other place in the world and would prefer to see content from that region. This is why you can set the **region** in Windows to wherever you are at the moment. All the while you can still have your system language in the language you are comfortable with. And then there are date, time and number formats, which can also be configured independent of region and language.

I'm frankly tired of seeing these issues in Microsoft applications, as these settings things have been in Windows for at least 20 years and I'm shocked that Microsoft developers still don't understand how these work.

**Comment by Otiel** [comment]
https://github.com/microsoft/WSL/releases/tag/2.1.3

> Use the display language for localization instead of the default locale (solves #11087,#11051)

**Comment by Florian-Thake** [comment]
> The problem here is that the application does **not** follow the system language, but rather the region. These are two different settings. A user can be located in Germany, Poland, France or any other place in the world and would prefer to see content from that region. This is why you can set the **region** in Windows to wherever you are at the moment. All the while you can still have your system language in the language you are comfortable with. And then there are date, time and number formats, which can also be configured independent of region and language.
> 
> And applications **should** follow system language. It's that simple.
> 
> I'm frankly tired of seeing these issues in Microsoft applications, as these settings have been in Windows for at least 20 years and I'm shocked that Microsoft developers **still** don't grasp this concept.

Yes, that is the bug here and must be fixed.

Furthermore there are not only the region and the display language but also the input method (keyboard layout) and the formatting options for currency, date and so on as well as the time zone. All of these settings must be changeable completely independently and must be respected by every application. 

> I disagree. If applications are translated, they should come in the language the user expects. You're using a generalisation to express your personal opinion. That does not help the quality of the application and only shifts the issue to a different group of people - namely the ones who want localised apps. If **you** want your system and apps to be in English, then you can do that via the system language settings.

Sure, that is one way.
But I claim that there are different kind of applications. There are those which main target audience are developers and administrators. And for those I claim, that there is only one working way for avoid misunderstandings and errors: English language. It does not make sense to translate technical keywords or phrases into some other language. It will only lead to subtle differences in the meaning and a lost in some details (best case) or even to big errors (worst case).
For example, to program VBA in Excel in any other language than English is just a pain. English is the common language for interchange. Imagine, everybody would program an API of a library, the source and all the comments in his/her native spoken language. The IT world would collapse entirely.
So, again, I claim, that for tools which are targeted for developers and administrators are best available only in English.
It will lead to global interchangeable results automatically.
This is not only my personal opinion but backed with reasonable points.

When it comes to personal opinion, I can add the following:
I not only want that application will follow the set system language. I want set the display language for each application individually.
As I said, tools for developing shall be in English for me, but my Word shall be in German as well as any movie app. Also, for movies, I don't want English or English subtitles but German for default, and so on.

**Comment by Florian-Thake** [comment]
> https://github.com/microsoft/WSL/releases/tag/2.1.3
> 
> > Use the display language for localization instead of the default locale (solves #11087,#11051)

That sounds promising. Thank you!

**Comment by OneBlue** [comment]
Solved with [2.1.3](https://github.com/microsoft/WSL/releases/tag/2.1.3)
